server/post_test.go

```diff
@@ -0,0 +1,258 @@
+				api.On("KVGet", tokenKey+"testUserID").Return(encryptedUserInfo, nil).Once()
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(nil, &model.AppError{Message: "error occurred getting the channel"})
+			},
+		},
+		{
+			name:          "Error getting channel members",
+			creator:       &model.User{Id: "testUserID"},
+			expectedError: "error occurred getting channel members",
+			setup: func() {
+				api.On("KVGet", tokenKey+"testUserID").Return(encryptedUserInfo, nil).Once()
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(&model.Channel{Id: "testChannelID", Type: model.ChannelTypeDirect}, nil)
+				api.On("GetChannelMembers", "testChannelID", 0, 100).Return(nil, &model.AppError{Message: "error occurred getting channel members"})
+			},
+		},
+		{
+			name:          "Channel members is nil",
+			creator:       &model.User{Id: "testUserID"},
+			expectedError: "returned members is nil",
+			setup: func() {
+				api.On("KVGet", tokenKey+"testUserID").Return(encryptedUserInfo, nil).Once()
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(&model.Channel{Id: "testChannelID", Type: model.ChannelTypeDirect}, nil)
+				api.On("GetChannelMembers", "testChannelID", 0, 100).Return(nil, nil)
+			},
+		},
+		{
+			name:          "Error creating the meeting",
+			creator:       &model.User{Id: "testUserID"},
+			expectedError: "error creating the meeting",
+			setup: func() {
+				api.On("KVGet", tokenKey+"testUserID").Return(encryptedUserInfo, nil).Once()
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(&model.Channel{Id: "testChannelID", Type: model.ChannelTypeDirect}, nil)
+				api.On("GetChannelMembers", "testChannelID", 0, 100).Return(model.ChannelMembers{}, nil)
+				client.On("CreateMeeting").Return(&msgraph.OnlineMeeting{}, errors.New("error creating the meeting"))
+			},
+		},
+		{
+			name:          "Error creating the meeting post",
+			creator:       &model.User{Id: "testUserID", Username: "testUsername"},
+			expectedError: "error creating the post",
+			setup: func() {
+				require.NoError(t, err)
+				api.On("KVGet", tokenKey+"testUserID").Return(encryptedUserInfo, nil).Once()
+				api.On("HasPermissionToChannel", "testUserID", "testChannelID", model.PermissionCreatePost).Return(true)
+				api.On("GetChannel", "testChannelID").Return(&model.Channel{Id: "testChannelID", Type: model.ChannelTypeDirect}, nil)
+				api.On("GetChannelMembers", "testChannelID", 0, 100).Return(model.ChannelMembers{}, nil)
+				api.On("CreatePost", mock.Anything).Return(nil, &model.AppError{Message: "error creating the post"})
```

Comment:
Let's stay typesafe for these tests. 
```go
mockPost := mock.AnythingOfType("*model.Post")
```

then 

```go
api.On("CreatePost", mockPost).Return(nil, &model.AppError{Message: "error creating the post"})
```

Do this for all instances.